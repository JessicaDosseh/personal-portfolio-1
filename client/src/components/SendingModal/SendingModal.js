import React from "react";
import PropTypes from "prop-types";
import styled, { css, keyframes } from "styled-components";

const moveTLBL = keyframes`
  0% {top: 0; left: 0;}
  50% {top: 50px; left: 50px; transform: rotate(90deg); opacity: 0.9;}
  100% {top: 100px; left: 0; transform: rotate(180deg);}
`;

const moveBRTR = keyframes`
  0% {top: 100px; left: 100px;}
  50% {top: 50px; left: 50px; transform: rotate(-90deg); opacity: 0.9;}
  100% {top: 0; left: 100px; transform: rotate(-180deg);}
`;

const moveBLTL = keyframes`
  0% {top: 100px; left: 0;}
  50% {top: 50px; left: 50px; transform: rotate(90deg); opacity: 0.9}
  100% {top: 0; left: 0; transform: rotate(180deg);}
`;

const moveTRBR = keyframes`
  0% {top: 0; left: 100px;}
  50% {top: 50px; left: 50px; transform: rotate(-90deg); opacity: 0.9}
  100% {top: 100px; left: 100px; transform: rotate(-180deg);}
`;

const Modal = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  padding: 20vh 5% 0;
  background: white;
  width: 100%;
  min-height: 100vh;
  z-index: 50;

  ${props =>
    props.darkmode &&
    css`
      background: #2f3234;
    `}

  div {
    height: 140px;
    width: 140px;
    margin: 0 auto;
    position: relative;

    div {
      height: 40px;
      width: 40px;
      position: absolute;
      border-radius: 10px;
    }
  }

  p {
    margin-top: 35px;
    display: block;
    width: 100%;
    text-align: center;

    ${props =>
      props.darkmode &&
      css`
        color: white;
      `}
  }
`;

const BoxOne = styled.div`
  background: #6936de;
  z-index: 5;
  animation-name: ${moveTLBL};
  animation-duration: 1.2s;
  animation-iteration-count: infinite;
  animation-timing-function: ease-in-out;
  animation-direction: alternate;
  animation-fill-mode: backwards;
  animation-delay: 0.1s;

  ${props =>
    props.darkmode &&
    css`
      background: black;
      opacity: 0.6;
    `}
`;

const BoxTwo = styled.div`
  background: #6936de;
  animation-name: ${moveBRTR};
  animation-duration: 1.2s;
  animation-iteration-count: infinite;
  animation-timing-function: ease-in-out;
  animation-direction: alternate;
  animation-fill-mode: backwards;
  animation-delay: 0.1s;

  ${props =>
    props.darkmode &&
    css`
      background: black;
      opacity: 0.6;
    `}
`;

const BoxThree = styled.div`
  background: #7950e0;
  animation-name: ${moveBLTL};
  animation-duration: 1.2s;
  animation-iteration-count: infinite;
  animation-timing-function: ease-in-out;
  animation-direction: alternate;
  animation-fill-mode: backwards;
  animation-delay: 0.1s;

  ${props =>
    props.darkmode &&
    css`
      background: black;
    `}
`;

const BoxFour = styled.div`
  background: #7950e0;
  animation-name: ${moveTRBR};
  animation-duration: 1.2s;
  animation-iteration-count: infinite;
  animation-timing-function: ease-in-out;
  animation-direction: alternate;
  animation-fill-mode: backwards;
  animation-delay: 0.1s;

  ${props =>
    props.darkmode &&
    css`
      background: black;
    `}
`;

const SendingModal = props => {
  return (
    <Modal darkmode={props.darkmode}>
      <div>
        <BoxOne darkmode={props.darkmode} />
        <BoxTwo darkmode={props.darkmode} />
        <BoxThree darkmode={props.darkmode} />
        <BoxFour darkmode={props.darkmode} />
      </div>
      <p>Sending</p>
    </Modal>
  );
};

SendingModal.propTypes = {
  darkmode: PropTypes.bool
};

export default SendingModal;
