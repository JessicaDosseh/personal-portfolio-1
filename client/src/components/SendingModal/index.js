export { default as SendingModal } from "./SendingModal";