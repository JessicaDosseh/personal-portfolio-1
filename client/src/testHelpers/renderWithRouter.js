import React from "react";
import { Router } from "react-router-dom";
import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { globalReducer, globalContext } from "../store/reducers/globalReducer";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(
      <StateProvider reducer={globalReducer} stateContext={globalContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default renderWithRouter;
