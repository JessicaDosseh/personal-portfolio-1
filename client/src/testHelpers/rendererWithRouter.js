import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { globalReducer, globalContext } from "../store/reducers/globalReducer";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <StateProvider reducer={globalReducer} stateContext={globalContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default rendererWithRouter;
