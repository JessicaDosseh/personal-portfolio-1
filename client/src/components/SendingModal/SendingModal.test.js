import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "../../testHelpers";

import SendingModal from "./SendingModal";

describe("<SendingModal />", () => {
  it("should render the SendingModal component without crashing", () => {
    const helpers = renderWithRouter(<SendingModal darkmode={false} />);
  });

  it("should match the snapshot of SendingModal", () => {
    const tree = rendererWithRouter(<SendingModal darkmode={false} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
