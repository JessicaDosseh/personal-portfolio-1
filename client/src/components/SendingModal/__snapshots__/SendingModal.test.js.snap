// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SendingModal /> should match the snapshot of SendingModal 1`] = `
<div
  className="sc-AykKC jRUWhM"
>
  <div>
    <div
      className="sc-AykKD dXsVUQ"
    />
    <div
      className="sc-AykKE dzgBFK"
    />
    <div
      className="sc-AykKF hmgoQC"
    />
    <div
      className="sc-AykKG iexOPT"
    />
  </div>
  <p>
    Sending
  </p>
</div>
`;
